src/corona_hakab_model/medical_state.py

```diff
@@ -9,12 +9,33 @@ class MedicalState(State, ABC):
     test_willingness: float
     detectable: bool
 
-    def __init__(self, *args, **kwargs):
-        self.susceptible = kwargs.pop('susceptible')
-        self.contagiousness = kwargs.pop('contagiousness')
-        self.test_willingness = kwargs.pop('test_willingness')
-        self.detectable = kwargs.pop('detectable')
-        super().__init__(*args, **kwargs)
+    def __init__(self, susceptible, contagiousness, test_willingness, detectable, **kwargs):
+        self.susceptible = susceptible
+        self.contagiousness = contagiousness
+        self.test_willingness = test_willingness
+        self.detectable = detectable
+        super().__init__(**kwargs)
+
+    @staticmethod
+    def build(mechanism, susceptible, contagiousness, test_willingness, detectable, **kwargs):
+        """
+        Builds the appropriate MedicalState state object from a Dictionary
+        @param detectable: bool, whether agent infection could theoretically be detected
+        @param test_willingness: float, probability that agent will be willing to be tested
+        @param contagiousness: float, probability of infecting others
+        @param susceptible: bool, whether agent can be infected
+        @param mechanism: str, Whether the medical state machine should be 'terminal' or 'stochastic'
+        @param kwargs: dict, name parameter is required
+        @return: an instance of a MedicalState child class
+        """
+        medical_state_type = mechanism
+
+        if medical_state_type == 'terminal':
+            return TerminalMedicalState(susceptible, contagiousness, test_willingness, detectable, **kwargs)
+        elif medical_state_type == 'stochastic':
+            return StochasticMedicalState(susceptible, contagiousness, test_willingness, detectable, **kwargs)
+        else:
+            raise KeyError("Medical state type invalid!")
```

Comment:
this should just be a dict in the future

Reply:
But it can’t be a dict. It is the instance that stores medical state and holds the methods to transition medical state. Am I missing something?

Reply:
```
type_cls = {'terminal':TerminalMedicalState, 'stochastic': StochasticMedicalState}
...
return type_cls[medical_state_type](...)
```